fencing/commands.c

```diff
@@ -1212,6 +1219,20 @@ stonith_device_remove(const char *id, gboolean from_cib)
         g_hash_table_remove(device_list, id);
         crm_info("Removed '%s' from the device list (%d active devices)",
                  id, g_hash_table_size(device_list));
+    } else if (!device->cib_registered) {
+        GListPtr gIter = NULL;
+        for (gIter = device->pending_ops; gIter != NULL;) {
+            async_command_t *cmd = gIter->data;
+            GListPtr next = gIter->next;
+
+            if (cmd->delay_id) {
+                crm_warn("Removal of device '%s' from cib purged delayed operation %s",
+                         device->id, cmd->action);
+                cmd->done_cb(0, -ENODEV, NULL, cmd);
+                device->pending_ops = g_list_delete_link(device->pending_ops, gIter);
+            }
+            gIter = next;
+        }
```

Comment:
I don't think we want this change. If the device is still API registered after being removed/disabled in the CIB, the device needs to stay active. The good news is that no one except our own regression tests ever use the API to register devices, so it shouldn't be a problem in practice.